sdk/src/store.rs

```diff
@@ -3129,7 +3219,11 @@ impl Store {
         output_stream: &mut dyn CAIReadWrite,
         reserve_size: usize,
     ) -> Result<Vec<u8>> {
-        let intermediate_output: Vec<u8> = Vec::new();
+        // make sure we can hold the intermediate stream before attempting
+        let intermediate_output: Vec<u8> = safe_vec(
```

Comment:
maybe factor this out into a function that does a safe large vec allocate and returns an API specific error?

Reply:
Actually I'd go a step further and have it return a stream, where that stream could be a temp file over some size if the file_io feature is available. Plus it could check for settings on size limits and file_backup options.